### ISSUE TITLE
Dropdown Component - Invalid aria-expanded attribute and invalid id referenced by for attribute

### ISSUE THREAD

**Issue by shunjieyu** [IBM/carbon-components-angular#1133]
## Detailed description

> Describe in detail the issue you're having.
- Our AVT team has found some violation via DAP tool that the aria-expanded attribute is invalid due to missing corresponding role, and also the id referenced by the 'for' attribute of a LABEL element is not a valid FORM INPUT element.

> Is this a feature request (new component, new icon), a bug, or a general issue?
- Bug

> Is this issue related to a specific component?
- Dropdown

> What did you expect to happen? What happened instead? What would you like to see changed?

> What browser are you working in?
- Chrome

> What version of the Carbon Design System are you using?
- V3.40.4

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
- IBM ADDI

**Comment by immanenipavankumar** [comment]
This defect causes our IBM product to be ACS (Accessibility Compliance System) "NOT Compliant" and would like a fix for this as soon as possible within a week.
